openshift-knative-operator/pkg/serving/extension.go

```diff
@@ -201,3 +206,29 @@ func (e *extension) fetchLoggingHost(ctx context.Context) string {
 	}
 	return route.Status.Ingress[0].Host
 }
+
+func (e *extension) cleanupDomainMapping(ctx context.Context, ks *operatorv1beta1.KnativeServing) error {
```

Comment:
To make this cheap I think we need to move it to the operator main as we did in the past with the other old resources we deleted. Otherwise we need a lister for all the object types.

Reply:
Yep, but checking if Deployment is present might not be sufficient to determine clean necessity. Is it a good idea to gate it based on Serving or S-O version?

Reply:
Or just simple `domainMappingCleaned bool`? Then it's executed once per new operator instance.

Reply:
You can either:
 - Keep it as is and have a mutex or some similar concept eg. sync.Once, making sure this runs once if no error happened.
 - add in the main and run it once there (with/without retries)